mmdet/core/evaluation/eval_hooks.py

```diff
@@ -10,17 +10,35 @@ class EvalHook(Hook):
     Attributes:
         dataloader (DataLoader): A PyTorch dataloader.
         interval (int): Evaluation interval (by epochs). Default: 1.
+        start (int, optional): Evaluation starting epoch. It enables evaluation
+            before the training starts if ``start`` <= the resuming epoch.
+            If None, whether to evaluate is merely decided by ``interval``.
+            Default: None.
+        **eval_kwargs: Evaluation arguments fed into the evaluate function of
+            the dataset.
     """
 
-    def __init__(self, dataloader, interval=1, **eval_kwargs):
+    def __init__(self, dataloader, interval=1, start=None, **eval_kwargs):
         if not isinstance(dataloader, DataLoader):
             raise TypeError('dataloader must be a pytorch DataLoader, but got'
                             f' {type(dataloader)}')
         self.dataloader = dataloader
         self.interval = interval
+        self.start = start
         self.eval_kwargs = eval_kwargs
+        self.initial_epoch_flag = True
+
+    def before_train_epoch(self, runner):
+        """Evaluate the model only at the start of training."""
+        if not self.initial_epoch_flag:
+            return
+        if self.start is not None and runner.epoch >= self.start:
+            self.after_train_epoch(runner)
```

Comment:
In `self.after_train_epoch`, the condition `self.every_n_epochs(runner, self.interval)` may not be satisfied in this case.

Reply:
Then it would depend on the interval to decide. For example, if interval=2 , start=1, and the resuming epoch =1, it is logically correct not to run evaluation before the training epoch.